internal/xds/translator/translator.go

```diff
@@ -577,3 +600,82 @@ func addXdsCluster(tCtx *types.ResourceVersionTable, args *xdsClusterArgs) error
 	}
 	return nil
 }
+
+const (
+	DefaultEndpointType EndpointType = iota
+	Static
+	EDS
+)
+
+func buildXdsUpstreamTLSCertSecret(tlsConfig ir.TLSBundle) *tlsv3.Secret {
+	// Build the tls secret
+
+	return &tlsv3.Secret{
+		Name: tlsConfig.GetXdsCertSecretName(), // GetBackendTLSCertSecretName(tlsConfig),
+		Type: &tlsv3.Secret_TlsCertificate{
+			TlsCertificate: &tlsv3.TlsCertificate{
+				CertificateChain: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.CertificateByte},
+				},
+				PrivateKey: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.PrivateKeyByte},
+				},
+			},
+		},
+	}
+}
+
+func buildXdsUpstreamTLSCASecret(tlsConfig ir.TLSBundle) *tlsv3.Secret {
+	// Build the tls secret
+
+	return &tlsv3.Secret{
+		Name: tlsConfig.GetXdsCaSecretName(), // GetBackendTLSCaSecretName(tlsConfig),
+		Type: &tlsv3.Secret_ValidationContext{
+			ValidationContext: &tlsv3.CertificateValidationContext{
+				TrustedCa: &corev3.DataSource{
+					Specifier: &corev3.DataSource_InlineBytes{InlineBytes: tlsConfig.CaCertificate},
+				},
+			},
+		},
+	}
+}
+
+func buildXdsUpstreamTLSSocketWthCert(tlsConfigs *ir.TLSBundle) (*corev3.TransportSocket, error) {
+
+	tlsCtx := &tlsv3.UpstreamTlsContext{
+		CommonTlsContext: &tlsv3.CommonTlsContext{
+			TlsCertificateSdsSecretConfigs: func() []*tlsv3.SdsSecretConfig {
+				if !tlsConfigs.Mtls() {
+					return nil
+				}
+				x := []*tlsv3.SdsSecretConfig{
+					{
+						Name:      tlsConfigs.GetXdsCertSecretName(),
+						SdsConfig: makeConfigSource(),
+					},
+				}
+				return x
+			}(),
+			ValidationContextType: &tlsv3.CommonTlsContext_ValidationContextSdsSecretConfig{
+				ValidationContextSdsSecretConfig: &tlsv3.SdsSecretConfig{
+					Name:      tlsConfigs.GetXdsCaSecretName(),
+					SdsConfig: makeConfigSource(),
+				},
+			},
+		},
+		Sni:                tlsConfigs.Hostname,
+		AllowRenegotiation: true,
```

Comment:
Is there a real use case for enabling renegotiation by default? This TLS feature received a lot of criticism over the years and was finally dropped in TLS1.3. 

Envoy docs: "TLS renegotiation is considered insecure and shouldn’t be used unless absolutely necessary."

The common use case that I'm familiar with is sending a client certificate post-handshake. Is that the use case? If so, maybe this can be enabled only when MTLS is configured?

Reply:
so keeping it true with mtls

Reply:
Maybe some of the maintainers can chime-in here. I'm still hesitant about enabling by default something that envoy docs are so opposed to ... @arkodg

Reply:
+1 to removing this

Reply:
set to false